guides/effectsize_exemplar_within.Rmd

```diff
@@ -31,93 +31,266 @@ format_num <- function(nums, sigdigits = 3) gsub("\\.$", "", formatC(nums, sigdi
 
 
 ### Subjects, conditions, and repetitions
-In this example, there are 10 subjects (`id` column). Because this is simulated data, we're using subject `id` to represent individual performance differences. Because within-subjects experiments partly account for individual differences, they often need far fewer subjects than between-subject designs. Repetitions also help reduce noise.
+In this example, there are 6 subjects (`subject` column).
 
 ```{r within-setup}
-set.seed(456) # make the output consistent
+set.seed(543) # make the output consistent
+SUBJECT_COUNT = 6
+
 data <- expand.grid(
-  id = rnorm(6, 5, 0.5), # individual differences
+  subject = paste('Subject', LETTERS[1:SUBJECT_COUNT]), # subject IDs
   layout = 0:1, # independent variable
   size = 0:2, # independent variable
   color = 0:3, # independent variable
   repetition = 1:20 # each subject runs in each condition multiple times
 )
 ```
 
+### Individual differences
+Not all subjects behave the same way. Some people might be tired, bad at the task, or just not trying very hard. These performance differences can't be directly measured, but they can substantially impact the results. We'll simulate these individual differences by giving each subject a random performance handicap.
+
+```{r}
+# build a table of subject performance multipliers
+individualDifferences <- tibble(subject = unique(data$subject))
+individualDifferences$handicap <- rnorm(SUBJECT_COUNT, 20, 4) # individual differences
+
+# put it in the main dataset
+data <- data %>% left_join(individualDifferences, by = "subject")
+```
+
+
 ### Simulate some noisy effects
 We'll simulate an experiment with a main effect of `layout` and `size` and an interaction between them. However, `color` and its interactions will not have an impact.
 
 ```{r within-simulate}
+
+# simulate the response times with a clean model
 data <- 
   data %>% 
   mutate(
   response_time = 
-    id + # additive individual differences
     layout * .4 + # main effect of layout
     size * .2 + # main effect of size
     color * 0 + 
-    layout * size * .5 + # 2-way interaction
+    layout * size * .6 + # 2-way interaction
     size * color * 0 + 
     layout * color * 0 + 
-    layout * size * color * 0 + 
-    rnorm(n()) # noise
+    layout * size * color * 0
 )
+
+# add some reponse noise
+data <- data %>% mutate(response_time = response_time + rnorm(n()))
+
+# add noise from individual handicaps
+data <- data %>% mutate(response_time = 30 + handicap*2 + response_time * handicap)
+
 ```
 
 Even though we used numbers to simulate the model, the independent variables and subject ID are all factors.
 ```{r within-factor}
 data <- 
   data %>% 
   mutate(
-    id = factor(id), 
+    subject = factor(subject), 
     layout = factor(layout), 
-    size = factor(size), 
     color = factor(color)
   )
 ```
 
```

Comment:
Half of this section concerns data generation. Would it make the exemplar more accessible if we move the data generation to an appendix?

Reply:
It's somewhat important that people understand how the per-subject noise impacts the simulations. But I've pulled all of the data generation into one large section.